src/eth1spec/spec.py

```diff
@@ -330,6 +342,126 @@ def signing_hash(tx: Transaction) -> Hash32:
     )
 
 
+def header_hash(header: Header) -> Hash32:
+    """
+    Computes the hash of a block Header.
+
+    Parameters
+    ----------
+    header : `eth1spec.eth_types.Header`
+        Header of interest.
+
+    Returns
+    -------
+    hash : `eth1spec.eth_types.Hash32`
+        Hash of the Header.
+    """
+    return crypto.keccak256(
+        rlp.encode(
+            (
+                header.parent_hash,
+                header.ommers_hash,
+                header.coinbase,
+                header.state_root,
+                header.transactions_root,
+                header.receipt_root,
+                header.bloom,
+                header.difficulty,
+                header.number,
+                header.gas_limit,
+                header.gas_used,
+                header.timestamp,
+                header.extra_data,
+                header.mix_digest,
+                header.nonce,
+            )
+        )
+    )
+
+
+def get_block_header_by_hash(hash: Hash32, chain: BlockChain) -> Header:
+    """
+    Fetches the block header by searching its hash.
+
+    Parameters
+    ----------
+    hash : `eth1spec.eth_types.Hash32`
+        hash of the header of interest.
+
+    chain : `eth1spec.eth_types.BlockChain`
+        History and current state.
+
+    Returns
+    -------
+    Header : `eth1spec.eth_types.Header`
+        Block header found by its hash.
+    """
+    for block in chain.blocks:
+        if header_hash(block.header) == hash:
+            return block.header
+    else:
+        raise ValueError("Could not find Header with hash {!r}".format(hash))
```

Comment:
>  Again, not really something specifically for this PR but more a style guide consistency comment for this whole repository.

Curious of the choice of `.format(...)` over f-strings.

Reply:
flake started complaining when I used f-strings. I could address this in a separate PR.

Reply:
Was it complaining inside of `tox`, or are you running it outside the build environment?

Reply:
Huh, they seemed to be working here: https://github.com/quilt/eth1.0-specs/blob/pyspec/tests/test_trie.py#L20

Reply:
It's `mypy` complaining for some reason, not flake8. I get the following when I run `tox -e py3`:
```
py3 run-test: commands[3] | mypy src tests setup.py --exclude 'tests/fixtures/*'
src/eth1spec/spec.py:403: error: On Python 3 '{}'.format(b'abc') produces "b'abc'", not 'abc'; use '{!r}'.format(b'abc') if this is desired behavior
```
-------
Upon investigating I realized that I need convert the hash value(bytes) to hex for the sake of readability.